### PR TITLE
record previous create org operation info when return to error page

### LINKED ISSUE #22833: Holding the values in form when the post request is failed with error
### Description

At first, I found that some vaules in the form were missing or changed to default values automaticly, when creating an org or a repo with invalid name.
e.g.
image

`Visibility` in this page is required so users will be aware of the change.
But if there are some no required params with this bug, user may post wrong data with no awareness of these changes.

Then I found func `AssignForm` in modules/web/middleware/binding.go which is used to assign form values back to the template data.
It seems we need to define same variable names between form data and template data?

e.g.
`Visibility` in this page is defined as `DefaultOrgVisibilityMode` in template data, but `visibility` in form data.

### Gitea Version

87261f3fb95da92d74d72d057fc0f1e9819f16a7

### Can you reproduce the bug on the Gitea demo site?

No

### Log Gist

_No response_

### Screenshots

_No response_

### Git Version

_No response_

### Operating System

_No response_

### How are you running Gitea?

build

### Database

None

### PR DESCRIPTION
**Pull request by sillyguodong**
Fixes: #22833 
Put user's previous operation params into context, for the page can get these params when re-rendering

### CODE REVIEW

File: templates/org/create.tmpl [hunk 1]
```diff
@@ -19,15 +19,15 @@
 						<span class="inline required field"><label for="visibility">{{.locale.Tr "org.settings.visibility"}}</label></span>
 						<div class="inline-grouped-list">
 							<div class="ui radio checkbox">
-								<input class="hidden enable-system-radio" tabindex="0" name="visibility" type="radio" value="0" {{if .DefaultOrgVisibilityMode.IsPublic}}checked{{end}}/>
+								<input class="hidden enable-system-radio" tabindex="0" name="visibility" type="radio" value="0" {{if or .DefaultOrgVisibilityMode.IsPublic (and .Err_OrgName .visibility.IsPublic)}}checked{{end}}/>
```
Comment (wolfogre) [inline templates/org/create.tmpl]: Maybe
```suggestion
								<input class="hidden enable-system-radio" tabindex="0" name="visibility" type="radio" value="0" {{if or .DefaultOrgVisibilityMode.IsPublic .visibility.IsPublic}}checked{{end}}/>
```
is enough.

Comment (sillyguodong) [reply templates/org/create.tmpl]: U are right👍🏻

File: templates/org/create.tmpl [hunk 2]
```diff
@@ -37,7 +37,7 @@
 						<label>{{.locale.Tr "org.settings.permission"}}</label>
 						<div class="inline-grouped-list">
 							<div class="ui checkbox">
-								<input class="hidden" type="checkbox" name="repo_admin_change_team_access" checked/>
+								<input class="hidden" type="checkbox" name="repo_admin_change_team_access" {{if or (not .Err_OrgName) .repo_admin_change_team_access}}checked{{end}}/>
```
Comment (wolfogre) [inline templates/org/create.tmpl]: There could be other errors, not only `Err_OrgName`.

Maybe
```suggestion
								<input class="hidden" type="checkbox" name="repo_admin_change_team_access" {{if not (eq .repo_admin_change_team_access false)}}checked{{end}}/>
```

So checked only if `.repo_admin_change_team_access` is nil (the new page) or is true (the posted page with it checked).

Not really sure if it works, could you please do some tests if you take it?

**Review by wolfogre — APPROVED** [review]

**Comment by zeripath** [comment]
hmm... 

1. is `.visibility` always set? I have a feeling not and if so this is likely to cause an NPE unless gotemplate have got round to changing the `or` to short-circuit (which as far as I know was still only in discussion)
2. If `.visibility` is not always set we should use its presence or not to determine whether to use the last submitted form value or not.
i.e. 

```html
<input class="hidden enable-system-radio" tabindex="0" name="visibility" type="radio" value="0" {{if .visibility}}{{if  .visibility.IsPublic}}checked{{end}}{{else if .DefaultOrgVisibilityMode.IsPublic}}checked{{end}}/>
```

3. Or better just change the rendering code to always set `ctx.Data["visibility"]` with the correct default environment variables and simplify to:

```html
<input class="hidden enable-system-radio" tabindex="0" name="visibility" type="radio" value="0" {{if  .visibility.IsPublic}}checked{{end}}/>
```

**Pull request closed without merging**